src/main/java/org/apache/sling/distribution/journal/kafka/KafkaEndpoint.java

```diff
@@ -36,4 +37,17 @@
     @AttributeDefinition(name = "Kafka Default API Timeout",
             description = "Kafka Default API Timeout in ms.")
     int kafkaDefaultApiTimeout() default 60000;
+    
+    @AttributeDefinition(name = "Security protocol",
+            description = "e.g. SASL_SSL")
+    SecurityProtocol securityProtocol() default SecurityProtocol.PLAINTEXT;
+    
+    @AttributeDefinition(name = "Sasl mechanism",
+            description = "e.g. PLAIN")
+    String saslMechanism() default "";
```

Comment:
The [default SASL mechanism](https://kafka.apache.org/documentation/#api) seems to be `GSSAPI`, we should use it here as well.

Reply:
I am not sure what happens if I set this if I do  not want to use sasl at all. If you know it will not affect the plain kafka case then I will change it to this.

Reply:
Given we set the Kafka default values by default, we won't affect the plain kafka use case.

Kafka default [consumer configs](https://kafka.apache.org/documentation/#consumerconfigs) and [producer configs](https://kafka.apache.org/documentation/#producerconfigs) for `sasl.jaas.config` is `null`. 

Kafka default [consumer configs](https://kafka.apache.org/documentation/#consumerconfigs) and [producer configs](https://kafka.apache.org/documentation/#producerconfigs) for `sasl.mechanism` is `"GSSAPI"`.